contracts/yield-source/YearnV2YieldSource.sol

```diff
@@ -55,23 +55,23 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable {
     /// @param _vault YearnV2 Vault in which the Yield Source will deposit `token` to generate Yield
     /// @param _token Underlying Token / Deposit Token
     function initialize(
-        address _vault,
-        address _token
+        IYVaultV2 _vault,
+        IERC20Upgradeable _token
     ) 
         public 
     {
-        require(vault == address(0), "!already initialized");
-        require(IYVaultV2(_vault).token() == _token, "!incorrect vault");
-        require(IYVaultV2(_vault).activation() != uint256(0), "!vault not initialized");
+        require(address(vault) == address(0), "!already initialized");
+        require(_vault.token() == address(_token), "!incorrect vault");
+        require(_vault.activation() != uint256(0), "!vault not initialized");
         // NOTE: Vaults from 0.3.2 to 0.3.4 have dips in shareValue
-        require(!areEqualStrings(IYVaultV2(_vault).apiVersion(), "0.3.2"), "!vault not compatible");
-        require(!areEqualStrings(IYVaultV2(_vault).apiVersion(), "0.3.3"), "!vault not compatible");
-        require(!areEqualStrings(IYVaultV2(_vault).apiVersion(), "0.3.4"), "!vault not compatible");
+        require(!areEqualStrings(_vault.apiVersion(), "0.3.2"), "!vault not compatible");
+        require(!areEqualStrings(_vault.apiVersion(), "0.3.3"), "!vault not compatible");
+        require(!areEqualStrings(_vault.apiVersion(), "0.3.4"), "!vault not compatible");
 
         vault = _vault;
         token = _token;
 
-        IERC20Upgradeable(_token).approve(_vault, type(uint256).max);
+        _token.safeApprove(address(_vault), type(uint256).max);
```

Comment:
Looks good to me; only concern I have is in the `_depositInVault()` function.

Someone could *possibly* lock deposits by flash depositing over and over again, running down the allowance to zero.

This can be mitigated by checking the allowance in `_depositInVault()` to make sure it is greater than the balance of the token.  If the allowance is insufficient, the allowance can be first zeroed then increased to max int again (for compatibility with USDT)

Reply:
very good points thanks